app/taskroutes.py

```diff
@@ -0,0 +1,148 @@
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.models.task import Task
+from app.models.goal import Goal
+from app import db
+from sqlalchemy import asc, desc
+import datetime, requests, os
+from dotenv import load_dotenv
+
+load_dotenv()
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"{cls.__name__} {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model
```

Comment:
There's nothing wrong with this implementation of `validate_model` but considering this is used in both `goalroutes` and `taskroutes`, I would consider pulling it out into its own file (perhaps under the `models` folder, something like `models/model_utils`).

The main benefit would be that if I was in, say `goalroutes` and I wanted to know what the `validate_model` does, I wouldn't expect to look in `taskroutes`, where it is. Kind of just part of making your code maintainable.